trick_source/codegen/Interface_Code_Gen/PrintAttributes.cpp

```diff
@@ -420,10 +420,16 @@ void PrintAttributes::printIOMakefile() {
 
     makefile_io_src.open("build/Makefile_io_src") ;
     makefile_io_src
-        << "TRICK_IO_CXXFLAGS += -std=c++11 -Wno-invalid-offsetof -Wno-old-style-cast -Wno-write-strings -Wno-unused-variable" << std::endl
+        << "TRICK_IO_CXXFLAGS += -Wno-invalid-offsetof -Wno-old-style-cast -Wno-write-strings -Wno-unused-variable" << std::endl
         << std::endl
         << "ifeq ($(IS_CC_CLANG), 0)" << std::endl
         << "    TRICK_IO_CXXFLAGS += -Wno-unused-local-typedefs -Wno-unused-but-set-variable" << std::endl
+        << "    ifeq ($(shell test $(GCC_MAJOR) -lt 6; echo $$?), 0)" << std::endl
+        << "        TRICK_IO_CXXFLAGS += -std=c++11" << std::endl
+        << "    endif" << std::endl
+        << "endif" << std::endl
+        << "ifeq ($(IS_CC_CLANG), 1)" << std::endl
+        << "    TRICK_IO_CXXFLAGS += -std=c++14" << std::endl
```

Comment:
Anyone can tell Trick to use clang to build all of the sim by setting `TRICK_CC` to `clang`, so we can't assume it's C++14 compatible. I sometimes do that to get better error messages when building a sim. I would do the same thing you did with gcc and add `--std=c++11`  only for versions that don't enable support for it by default.